tests/tests_geomstats/test_functions.py

```diff
@@ -0,0 +1,44 @@
+"""Unit tests for the functions manifolds."""
+
+import math
+import warnings
+
+import numpy as np
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.functions import SinfSpace
+
+
+class TestFunctionsSinf(geomstats.tests.TestCase):
+    def setup_method(self):
+        self.domain = gs.linspace(-math.pi, math.pi)
```

Comment:
@kiranvad I believe replacing this by:
`gs.linspace(-gs.pi, gs.pi, num-50)` would solve the remaining issues.

Reply:
Thanks for the suggestion @ninamiolane 
I fixed this but the tests still fail with a pytorch backend because `gs.vectorize` thrown an error. I suspect this is because I make use of the `signature` parameter that exists in NumPy but not in PyTorch. I am not sure how to fix this, so for now I made the tests only for NumPy and autograd.